test/extensions/clusters/aggregate/cluster_test.cc

```diff
@@ -131,6 +134,24 @@ class AggregateClusterTest : public Event::TestUsingSimulatedTime, public testin
     lb_ = lb_factory_->create(lb_params_);
   }
 
+  Stats::Gauge& getCircuitBreakersStatByPriority(std::string priority, std::string stat) {
+    std::string stat_name_ = "circuit_breakers." + priority + "." + stat;
+    Stats::StatNameManagedStorage statStore(stat_name_,
+                                            cluster_->info()->statsScope().symbolTable());
+    return cluster_->info()->statsScope().gaugeFromStatName(statStore.statName(),
+                                                            Stats::Gauge::ImportMode::Accumulate);
+  }
+
+  void assertResourceManagerStat(ResourceLimit& resource, Stats::Gauge& remaining,
+                                 Stats::Gauge& open, bool expected_can_create,
+                                 unsigned int expected_count, unsigned int expected_remaining,
+                                 unsigned int expected_open) {
+    EXPECT_EQ(expected_can_create, resource.canCreate());
+    EXPECT_EQ(expected_count, resource.count());
+    EXPECT_EQ(expected_remaining, remaining.value());
+    EXPECT_EQ(expected_open, open.value());
```

Comment:
I expect the reviewer will want the metrics removed from the test.
That way, they can change the behaviour of circuit breakers in the future, and it won't break tests here.

You're fine to leave it in if you really really like them, just letting you know. <3

Reply:
I found the defaults in `source/common/upstream/upstream_impl.cc` 
https://github.com/envoyproxy/envoy/blob/main/source/common/upstream/upstream_impl.cc#L2070-L2075

```c++
  uint64_t max_connections = 1024;
  uint64_t max_pending_requests = 1024;
  uint64_t max_requests = 1024;
  uint64_t max_retries = 3;
  uint64_t max_connection_pools = std::numeric_limits<uint64_t>::max();
  uint64_t max_connections_per_host = std::numeric_limits<uint64_t>::max();
```

but can't think of a way to mirror those to establish the defaults in the tests 
so that if they were changed later that change would be reflected in the tests